datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAttachmentViewHolderFactory.kt

```diff
@@ -124,8 +124,23 @@ internal object QuestionnaireItemAttachmentViewHolderFactory :
         deleteButton.isEnabled = !isReadOnly
       }
 
-      private fun displayInitialPreview() {
-        questionnaireItemViewItem.answers.firstOrNull()?.valueAttachment?.let { attachment ->
+      private fun displayInitialPreviewOrClear() {
+        val answer = questionnaireItemViewItem.answers.firstOrNull()
+
+        // When there are several Attachment widgets, this prevents a bug when the user uploads a
+        // file then shows a preview on the first widget, the last widget will also have the same
+        // preview but with no appended answer. So we should always clear the preview when there is
+        // no answer.
+        // This happens because RecyclerView is recycling a widget that already displays the
+        // preview.
```

Comment:
I think this is a bit too verbose. also, this bug isn't to do with "first" or "last", it's just the fact that some views are recycled.

I'd suggest simply:

```suggestion
        // Clear preview if there is no answer to prevent showing old previews in views that have been recycled.
```

Reply:
sure